### DISCUSSION TITLE
How to get local video view layer iOS?

### DISCUSSION THREAD

**Discussion by tiwari1amrit** [ant-media/Ant-Media-Server#5253] [General]
Hi there,
How to show a published video view to another view in iOS Swift? Currently, my application flow is that lets `Screen A `will publish the  user local video for video conferencing, and after connecting I have to move to another view controller lets say `Screen B`, So is there any proper way to get that video layer from `AntMediaClient`? 
Currently, What I am doing is that
I am joining room on `Screen A` as
   ```
     let url = Urls.urlForPlay()
     self.conferenceClient.setWebSocketServerUrl(url: url)
     self.conferenceClient.setLocalView(container: self.videoView, mode: .scaleAspectFill)
     self.conferenceClient.joinRoom(roomId: roomId)
```
and after successfully connected, I am passing conferenceClient object to `Screen B` and I want to show that `self.videoView` on screen B. Is there any possible way through `AntMediaClient`

**Comment by tiwari1amrit** [comment]
I can simply pass the view layer from `Screen A` to `Screen B` but can't able to change video mode. Please provide me better solution.

**Comment by mekya** [comment]
Hi @tiwariammit,

Thank you for the question. What exactly do you want to achieve? Because it does not make too much sense for me to give the `ConferenceClient` object to a different screen. I mean if I know what exactly you want to achieve, I may provide a better answer. 

As an answer to your question, I think the best way is to use `RTCVideoTrack` to play the stream as in the playing the streams in [ConferenceController](https://github.com/ant-media/WebRTC-iOS-SDK/blob/master/WebRTC-Sample-App/ConferenceViewController.swift#LL201C1-L202C1). 
On the other hand, [`localVideoTrack`](https://github.com/ant-media/WebRTC-iOS-SDK/blob/master/WebRTCiOSSDK/api/webrtc/WebRTCClient.swift#L27) in the SDK is not exposed.  As a solution, we can expose the `RTCVideoTrack` in two ways. 
The first one is with the getter and the second one is via the delegate method. You can  

Just let me know what you want to achieve and how to proceed. 

Cheers
Oguz

**Reply by tiwari1amrit** [reply]
Hi Mekya,
Thank you for getting back to me. 
My question is that I got two views for the video conference which I attached below. 
On the `join meeting screen`, I joined meeting using 

```
self.conferenceClient.setWebSocketServerUrl(url: url)
self.conferenceClient.setLocalView(container: self.videoView, mode: .scaleAspectFit)
self.conferenceClient.joinRoom(roomId: roomId)
self.conferenceClient.publish(streamId: publisherStreamId)
```
which will publish the local video after connecting I have to move the `video conferencing screen`. On this screen, I have to show my local view which I gave as 

`self.conferenceClient.setLocalView(container: self.videoView, mode: .scaleAspectFill)
`
So the view will manage frame as `videoView` frame, currently, I am just passing `videoView` to  `video conferencing screen` and on this new controller I added videoView layer to new view as

```
self.previousLocalVideoView.frame = self.localVideoView.bounds
self.previousLocalVideoView.backgroundColor = .red
self.localVideoView.layer.addSublayer(self.previousLocalVideoView.layer)
```
My issue is that the conferenceClient video can't render it view again which will get a frame issue, I have attached a screenshot. 
**Note**: it may raise a question that `previousLocalVideoView` didn't take `localVideoView` frame, so I have added background color to remove your confusion. Another method is that creating new instance of AntMediaClient, but I can't able to off the speaker because turn off speaker function is a static method which will turn off all speaker. 

`    public static func speakerOff()
`
So, what will be the best solution to solve this issue? 

<img width="921" alt="Screenshot 2023-06-17 at 11 42 36 am" src="https://github.com/ant-media/Ant-Media-Server/assets/22368948/f901b1a6-3d22-4a86-9c78-bea91355c1d4">

<img width="1036" alt="Screenshot 2023-06-17 at 1 06 18 pm" src="https://github.com/ant-media/Ant-Media-Server/assets/22368948/9f668a43-92e1-47e3-914a-f3b0e82412c2">

**Comment by tiwari1amrit** [comment]
Hi Mekya, 
I solved it by managing the view frame. Thank you for your response.
